python/practice/python_calculator/project/test_calc.py

```diff
@@ -0,0 +1,192 @@
+            ["987654.321", 987654.321, "integer part with 6 digits"],
+            ["00.1", 0.1, "floating-point with leading zeroes"],
+            ["1.1000", 1.1, "floating-point with trailing zeroes"],
+            ["1.a", None, "floating-point with letter in fractional part"],
+            ["1.2e+3", None, "exponential representation"],
+            ["1.2.3", None, "floating-point with 2 dots"],
+        )
+
+        self.run_test_cases(cases)
+
+    def test_operators(self):
+        cases = (
+            ["1+2", 3, "expression with single +"],
+            ["1+22+333", 356, "expression with chain of + operators"],
+            ["9-11", -2, "expression with single - and negative result"],
+            ["11-9", 2, "expression with single - and positive result"],
+            [
+                "58-304-0",
+                -246,
+                "expression with chain of - operators (with 0)",
+            ],
+            ["9-1-1-0.5-1.5", 5, "expression with chain of - operators"],
+            ["3*11", 33, "expression with single *"],
+            [
+                "2*9*100*7",
+                12600,
+                "expression with chain of * operators (integers only)",
+            ],
+            [
+                "2.0*9.0*100.0*7.0",
+                12600,
+                "expression with chain of * operators (floating-points only)",
+            ],
+            ["49/7", 7, "expression with single /"],
+            ["28/2/2/2", 3.5, "expression with chain of / operators"],
+            [
+                "8/4/2",
+                1,
+                "expression with chain of / operators and integer result",
+            ],
+            ["5/0.5", 10, "expression with / operator and floating-point"],
+        )
+
+        self.run_test_cases(cases)
+
+    def test_operators_priority(self):
+        cases = (
+            ["2+5-1", 6, "operators with equal priority: +, -"],
+            ["2-5+1", -2, "operators with equal priority: -, +"],
+            ["4*2*3/11", 2.18181818, "operators with equal priority: *, /"],
+            ["4/2/3*11", 7.33333333, "operators with equal priority: /, *"],
+            ["3-9+8*2", 10, "operators with mixed priority: -, +, *"],
+            ["44*150-45/5", 6591, "operators with mixed priority: *, -, /"],
+            [
+                "6+400/500-1*1",
+                5.8,
+                "operators with mixed priority: +, /, -, *",
+            ],
+            ["1/2-1/2", 0, "operations with mixed priority: /, -, /"],
+            ["15+2*6-8", 19, "operations with mixed priority: +, *, -"],
+            [
+                "2-503*8+1/7",
+                -4021.85714286,
+                "operations with mixed priority: -, *, +, /",
+            ],
+            [
+                "5*2-6/3+5-7-9*0",
+                6,
+                "operators with mixed priority and 0 multipliplication",
+            ],
+        )
+
+        self.run_test_cases(cases)
+
+    def test_parenthesis(self):
```

Comment:
Надо везде поправить parenthesis на parentheses, где имеется в виду множественное число.

```suggestion
    def test_parentheses(self):
```